lib/puppet/provider/zone/solaris.rb

```diff
@@ -37,43 +53,42 @@ def self.instances
   # file or string.
   def configure
 
-    if @resource[:archive].nil? && @resource[:zonecfg_export].nil?
+    if @resource[:zonecfg_archive].nil? && @resource[:zonecfg_export].nil?
       raise Puppet::Error, "No configuration resource is defined."
     end
 
```

Comment:
This whole section is inconsistent; also pull in "22126555 Agent reporting syntax error while creating a zone"